Automatically Update Container Image Using Github Actions Workflow

mkell43 opened the pull request:
Howdy,

After forking your repo I went about setting things up to make sure they were appropriately linted and updated automatically. While I'm still in the process of setting up the workflows to run the checks on the images that are created, I thought you may be interested in pulling my workflow to centralize the process for ensuring that updates are applied to the container on a regular basis.

It only pushes to the Github registry which might not be your thing as you're currently using Docker Hub for hosting your container images. This could work in tandem or be edited to push to Docker Hub by referencing the documentation for `docker/build-push-action@v1` at : https://github.com/docker/build-push-action/tree/releases/v1#example-usage .

A quick breakdown of what's in the workflow file:
- Runs every day at noon UTC.
- Sets the version tag by referencing the Monero branch being used.
- Builds and pushes to the Github registry without needing any additional configuration to the repository settings or creating secrets.

sethsimmons commented:
Hey there,

Sorry for the delay, life has been crazy lately and this fell off my radar.

Merging now and will create CI to push to Docker Hub as well when I have the time.

Thanks for pushing back upstream 👍

sethsimmons commented:
@mkell43 I sort of fell down the rabbit-hole of Github Actions...

Feel free to snatch any of my updates, I added in pushes to Docker Hub and migrated to a cached build using registry, build-push-action@v2, and created a new action to run anytime the Dockerfile is updated.

mkell43 commented:
> Sorry for the delay, life has been crazy lately and this fell off my radar.

No worries! I started a new job and moved into a new home this month, so I'm well acquainted with life getting crazy and things dropping off.

>@mkell43 I sort of fell down the rabbit-hole of Github Actions... 

They're pretty great. If you haven't already, checkout [act](https://github.com/nektos/act). It allows you to test GH actions locally. There are some limitations, but overall it's a great way to catch things without having to get your repo's history messy.

> Feel free to snatch any of my updates, I added in pushes to Docker Hub and migrated to a cached build using registry, build-push-action@v2, and created a new action to run anytime the Dockerfile is updated.

I'll take a look this weekend. I'm doing a bunch of work with GH actions for work and w/ my personal stuff. Feel free to hit me up on Telegram with the same username if you want to talk about 'em or swap knowledge. (Disclaimer: Life is still crazy and I spend most of my time in Slack & not Telegram, but I'll catch your message/s eventually.)
